### DISCUSSION TITLE
How do you manage Tekton components installation and lifecycle management

### DISCUSSION THREAD

**Discussion by vdemeester** [tektoncd/community#1054] [Polls]
This poll is to gather feedback on how the community manages tekton components installation as well as their lifecycle (upgrade, …)

The idea is to figure out how, in general, our community manages Tekton components lifecycle so that we can focus on the right thing related to components lifecycle management.

Note on the "options", if you are using a `gitops`-based _flow_ to manager components lifecycle, please either choose other options or the "underlying" mechanism used (e.g. argocd with kustomize would go into "A custom setup based of `kustomize`", …).

**Comment by PoliM** [comment]
Currently we create our own Helm charts from the YAML files that were released. Very simplified: diff the manifests to see what changed and adjust the charts. It doesn't support all the features, just the ones we need. But to be honestly, every time I do this I wonder if I should switch to either a kustomize based process or check again if the operator would work in a cluster without direct internet access (last time I checked was a long time ago, so maybe it could).

**Comment by adambkaplan** [comment]
Shipwright's operator has an API dependency on the Tekton Operator - we use the `TektonConfig` API to configure Pipelines if we see that Tekton hasn't been deployed on the cluster. We also use labels on Tekton's CRDs to determine if Pipelines is deployed at a supported version.

**Comment by vdemeester** [comment]
Alright, I wonder how much this is a "close-to-truth" view of the community. 
I don't think this poll really reaches out the "end-users" as 10 votes is a relative small amount. In addition, I wonder how much of Red Hatter answer (that would bias the result a bit as we are doing "everything operator" 🙃.

However, I think it shows there is some value in having the operator, upstream, even without [OLM](https://github.com/operator-framework/operator-lifecycle-manager).

**Reply by vdemeester** [reply]
I *may* close this poll and submit a "better" one on the mailing list, hoping to reach more users that way.

**Reply by vdemeester** [reply]
The poll is available here : https://forms.gle/kXBfr59Bv3oJaxWM6
